components/mdns/tests/unit_test/main/test_mdns.c

```diff
@@ -142,12 +143,154 @@ TEST(mdns, query_api_fails_with_expected_err)
+    addr.next = NULL;
+    mdns_result_t *results = NULL;
+    test_case_uses_tcpip();
+    TEST_ASSERT_EQUAL(ESP_OK, esp_event_loop_create_default());
+    TEST_ASSERT_EQUAL(ESP_OK, mdns_init() );
+    TEST_ASSERT_EQUAL(ESP_OK, mdns_hostname_set(MDNS_HOSTNAME));
+    TEST_ASSERT_EQUAL(ESP_OK, mdns_delegate_hostname_add(MDNS_DELEGATE_HOSTNAME, &addr) );
+
+    TEST_ASSERT_EQUAL(ESP_OK, mdns_service_add_for_host(MDNS_INSTANCE, MDNS_SERVICE_NAME, MDNS_SERVICE_PROTO, MDNS_DELEGATE_HOSTNAME, MDNS_SERVICE_PORT, NULL, 0));
+    yield_to_all_priorities();  // Make sure that mdns task has executed to add the hostname
+    TEST_ASSERT_EQUAL(ESP_OK, mdns_lookup_delegated_service(NULL, MDNS_SERVICE_NAME, MDNS_SERVICE_PROTO, 1, &results));
+    TEST_ASSERT_NOT_EQUAL(NULL, results);
+    TEST_ASSERT_EQUAL_STRING(MDNS_INSTANCE, results->instance_name);
+    TEST_ASSERT_EQUAL_STRING(MDNS_SERVICE_NAME, results->service_type);
+    TEST_ASSERT_EQUAL(MDNS_SERVICE_PORT, results->port);
+    TEST_ASSERT_EQUAL(NULL, results->txt);
+    mdns_query_results_free(results);
+
+    // Update service properties: port
+    TEST_ASSERT_EQUAL(ESP_OK, mdns_service_port_set_for_host(NULL, MDNS_SERVICE_NAME, MDNS_SERVICE_PROTO, MDNS_DELEGATE_HOSTNAME, MDNS_SERVICE_PORT + 1));
+    yield_to_all_priorities();  // Make sure that mdns task has executed to add the hostname
+    TEST_ASSERT_EQUAL(ESP_OK, mdns_lookup_delegated_service(NULL, MDNS_SERVICE_NAME, MDNS_SERVICE_PROTO, 1, &results));
+    TEST_ASSERT_NOT_EQUAL(NULL, results);
+    TEST_ASSERT_EQUAL(MDNS_SERVICE_PORT + 1, results->port);
+    mdns_query_results_free(results);
+
+    // Update service properties: instance
+    TEST_ASSERT_EQUAL(ESP_OK, mdns_service_instance_name_set_for_host(NULL, MDNS_SERVICE_NAME, MDNS_SERVICE_PROTO, MDNS_DELEGATE_HOSTNAME, MDNS_INSTANCE "1" ));
+    yield_to_all_priorities();  // Make sure that mdns task has executed to add the hostname
+    TEST_ASSERT_EQUAL(ESP_OK, mdns_lookup_delegated_service(MDNS_INSTANCE "1", MDNS_SERVICE_NAME, MDNS_SERVICE_PROTO, 1, &results));
+    TEST_ASSERT_NOT_EQUAL(NULL, results);
+    TEST_ASSERT_EQUAL_STRING(MDNS_INSTANCE "1", results->instance_name);
+    mdns_query_results_free(results);
+
+    // Update service properties: txt
+    mdns_txt_item_t txt_data[] = {
+        {"key1", "esp32"},
+        {"key2", "value"},
+    };
+    const size_t txt_data_cout = sizeof(txt_data) / sizeof(txt_data[0]);
+    TEST_ASSERT_EQUAL(ESP_OK, mdns_service_txt_set_for_host(NULL, MDNS_SERVICE_NAME, MDNS_SERVICE_PROTO, MDNS_DELEGATE_HOSTNAME, txt_data, txt_data_cout));
+    yield_to_all_priorities();  // Make sure that mdns task has executed to add the hostname
+    TEST_ASSERT_EQUAL(ESP_OK, mdns_lookup_delegated_service(NULL, MDNS_SERVICE_NAME, MDNS_SERVICE_PROTO, 1, &results));
+    TEST_ASSERT_NOT_EQUAL(NULL, results);
+    TEST_ASSERT_NOT_EQUAL(NULL, results->txt);
+    // compare txt values by keys
+    size_t misses = 0;
+    for (int i = 0; i < txt_data_cout; ++i)
```

Comment:
Shouldn't we iterate on results?

Reply:
could be that you've reviewed the old version of the file?

Reply:
oh, I see. the same code for the delegated services. will fix

Reply:
Updated the same way, thanks for noticing!
And thanks for the review, I'll merge it to proceed with the actual API fix.